src/asciidoctorWebViewConverter.ts

```diff
@@ -49,4 +188,190 @@ export class AsciidoctorWebViewConverter {
+    }
+    if (details.length > 0) {
+      return `<div class="details">
+${details.join('\n')}
+</div>`
+    }
+    return ''
+  }
+
+  private generateManPageHeader (node) {
+    const tocContent = node.hasSections() && node.hasAttribute('toc') && node.hasAttribute('toc-placement', 'auto')
+      ? `<div id="toc" class="${node.getAttribute('toc-class', 'toc')}">
+<div id="toctitle">${node.getAttribute('toc-title')}</div>
+${node.getConverter().convert(node, 'outline')}
+</div>`
+      : ''
+    return `<h1>${node.getDoctitle()} Manual Page</h1>
+${tocContent}
+${node.hasAttribute('manpurpose') ? this.generateManNameSection(node) : ''}`
+  }
+
+  private generateManNameSection (node) {
+    let mannameTitle = node.getAttribute('manname-title', 'Name')
+    const nextSection = node.getSections()[0]
+    if (nextSection && nextSection.getTitle() === nextSection.getTitle().toUpperCase()) {
+      mannameTitle = mannameTitle.toUpperCase()
+    }
+    const mannameIdAttr = node.getAttribute('manname-id') ? ` id="${node.getAttribute('manname-id')}"` : ''
+    return `<h2${mannameIdAttr}>${mannameTitle}</h2>
+  <div class="sectionbody">
+    <p>${node.getAttribute('mannames').join(', ')} - ${node.getAttribute('manpurpose')}</p>
+  </div>`
+  }
+
+  private getBodyCssClasses (node) {
+    const classes = [
+      'vscode-body',
+      this.config.scrollBeyondLastLine ? 'scrollBeyondLastLine' : undefined,
+      this.config.wordWrap ? 'wordWrap' : undefined,
+      this.config.markEditorSelection ? 'showEditorSelection' : undefined,
+    ]
+    const sectioned = node.hasSections()
+    if (sectioned && node.isAttribute('toc-class') && node.isAttribute('toc') && node.isAttribute('toc-placement', 'auto')) {
+      classes.push(
+        node.getDoctype(),
+        node.getAttribute('toc-class'),
+        `toc-${node.getAttribute('toc-position', 'header')}`
+      )
+    } else {
+      classes.push(node.getDoctype())
+    }
+    if (node.isRole()) {
+      classes.push(node.getRole())
+    }
+    return classes
+      .filter((cssClass) => cssClass !== undefined)
+      .join(' ')
+  }
+
+  private getSettingsOverrideStyles (config: AsciidocPreviewConfiguration): string {
+    return [
+      config.fontFamily ? `--asciidoc-font-family: ${config.fontFamily};` : '',
+      isNaN(config.fontSize) ? '' : `--asciidoc-font-size: ${config.fontSize}px;`,
+      isNaN(config.lineHeight) ? '' : `--asciidoc-line-height: ${config.lineHeight};`,
+    ].join(' ')
+  }
+
+  private extensionResourcePath (mediaFile: string): string {
+    const webviewResource = this.editor.webview.asWebviewUri(vscode.Uri.joinPath(this.context.extensionUri, 'dist', mediaFile))
+    return webviewResource.toString()
+  }
+
+  private getStyles (resourceProvider: WebviewResourceProvider, resource: vscode.Uri, config: AsciidocPreviewConfiguration, state?: any): string {
+    // QUESTION: should we support `stylesdir` and `stylesheet` attributes?
```

Comment:
I think a lot of users want to be able to add custom CSS have a preview that supports this. I at least often want to add a small amount of custom styling and I often have a base folder with my own Asciidoctor stylesheet (since there's no facility to incrementally add styles in Asciidoctor) which sets the theme and is something I used in many repos/documents.

Reply:
Now that we do not rely exclusively on the `stylesheet` and `stylesdir` attributes, we can actually provide a mechanism or an option to add a stylesheet in addition to the built-in styles.